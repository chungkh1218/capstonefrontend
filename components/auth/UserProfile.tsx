/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

// import React, { Component } from "react";
import * as React from "react";
import { Component } from "react";
import {
  StyleSheet,
  Text,
  Dimensions,
  ImageBackground,
  Alert
} from "react-native";
import { Button } from "react-native-elements";
import { IAuthUser } from "../../models/models";
import { connect } from "react-redux";
import { logoutUser } from "../../redux/actions/AuthAction";
import { IRootState } from "../../redux/store";

interface IUserProfileProps {
  navigator: any;
  user: IAuthUser;
  logout: () => void;
}

class UserProfile extends Component<IUserProfileProps> {
  constructor(props: IUserProfileProps) {
    super(props);
  }

  handleLogout = () => {
    this.props.logout();
    if (this.props.user.isAuthenticated) {
      Alert.alert("Authenication", "You are signed out");
    }
    this.props.navigator.popToRoot({
      animated: true,
      animationType: "fade"
    });
  };

  render() {
    return (
      <ImageBackground
        source={require("../../src/img/Logout_BG.png")}
        style={styles.container}
      >
        <Text style={styles.header}>
          Hi, {this.props.user.username}. {"\n"} Welcome to Property Prince.
          {"\n"}
          We can you to find your ideal property investment here.
        </Text>

        <Text style={styles.description}>
          Email:
          {this.props.user.email}
        </Text>

        <Button
          title="Logout!"
          onPress={this.handleLogout}
          buttonStyle={{
            borderRadius: 24,
            backgroundColor: "#FF7600",
            marginHorizontal: 0,
            width: Dimensions.get("window").width * 0.35,
            margin: 8,
            padding: 12
          }}
        />
      </ImageBackground>
    );
  }
}

const mapStateToProps = (state: IRootState) => {
  return {
    // user: state.auth.user
  };
};

const mapDispatchToProps = (dispatch: any) => {
  return {
    logout: () => {
      dispatch(logoutUser());
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserProfile);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: Dimensions.get("window").width,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "white"
  },
  header: {
    fontSize: 16,
    lineHeight: 24,
    color: "black",
    textAlign: "center",
    margin: 10
  },
  description: {
    fontSize: 12,
    color: "grey",
    textAlign: "center",
    margin: 10
  }
});
