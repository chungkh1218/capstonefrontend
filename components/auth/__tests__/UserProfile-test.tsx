import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";

import UserProfile from "../UserProfile";

configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  navigator: "",
  user: {}
};

describe("Testing UserProfile", () => {
  it("renders as expected", () => {
    const wrapper = shallow(
      <UserProfile
        navigator
        user={{
          username: "peter",
          email: "pet@pet.com",
          user_id: 1,
          isAuthenticated: true
        }}
      />,
      {
        context: { store: mockStore(initialState) }
      }
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
