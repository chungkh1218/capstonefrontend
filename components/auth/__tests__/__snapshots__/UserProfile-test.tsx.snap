// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserProfile renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile
    logout={[Function]}
    navigator={true}
    user={
      Object {
        "email": "pet@pet.com",
        "isAuthenticated": true,
        "user_id": 1,
        "username": "peter",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "lineHeight": 24,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Hi, 
          peter
          . 
          

           Welcome to Property Prince.
          

          We can you to find your ideal property investment here.
        </Component>,
        <Component
          style={
            Object {
              "color": "grey",
              "fontSize": 12,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Email:
          pet@pet.com
        </Component>,
        <Button
          buttonStyle={
            Object {
              "backgroundColor": "#FF7600",
              "borderRadius": 24,
              "margin": 8,
              "marginHorizontal": 0,
              "padding": 12,
              "width": 262.5,
            }
          }
          onPress={[Function]}
          title="Logout!"
        />,
      ],
      "source": Object {
        "testUri": "../../../src/img/Logout_BG.png",
      },
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "width": 750,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Hi, ",
            "peter",
            ". ",
            "
",
            " Welcome to Property Prince.",
            "
",
            "We can you to find your ideal property investment here.",
          ],
          "style": Object {
            "color": "black",
            "fontSize": 16,
            "lineHeight": 24,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          "Hi, ",
          "peter",
          ". ",
          "
",
          " Welcome to Property Prince.",
          "
",
          "We can you to find your ideal property investment here.",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Email:",
            "pet@pet.com",
          ],
          "style": Object {
            "color": "grey",
            "fontSize": 12,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          "Email:",
          "pet@pet.com",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonStyle": Object {
            "backgroundColor": "#FF7600",
            "borderRadius": 24,
            "margin": 8,
            "marginHorizontal": 0,
            "padding": 12,
            "width": 262.5,
          },
          "onPress": [Function],
          "title": "Logout!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "lineHeight": 24,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Hi, 
            peter
            . 
            

             Welcome to Property Prince.
            

            We can you to find your ideal property investment here.
          </Component>,
          <Component
            style={
              Object {
                "color": "grey",
                "fontSize": 12,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Email:
            pet@pet.com
          </Component>,
          <Button
            buttonStyle={
              Object {
                "backgroundColor": "#FF7600",
                "borderRadius": 24,
                "margin": 8,
                "marginHorizontal": 0,
                "padding": 12,
                "width": 262.5,
              }
            }
            onPress={[Function]}
            title="Logout!"
          />,
        ],
        "source": Object {
          "testUri": "../../../src/img/Logout_BG.png",
        },
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
          "width": 750,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Hi, ",
              "peter",
              ". ",
              "
",
              " Welcome to Property Prince.",
              "
",
              "We can you to find your ideal property investment here.",
            ],
            "style": Object {
              "color": "black",
              "fontSize": 16,
              "lineHeight": 24,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "Hi, ",
            "peter",
            ". ",
            "
",
            " Welcome to Property Prince.",
            "
",
            "We can you to find your ideal property investment here.",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Email:",
              "pet@pet.com",
            ],
            "style": Object {
              "color": "grey",
              "fontSize": 12,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "Email:",
            "pet@pet.com",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonStyle": Object {
              "backgroundColor": "#FF7600",
              "borderRadius": 24,
              "margin": 8,
              "marginHorizontal": 0,
              "padding": 12,
              "width": 262.5,
            },
            "onPress": [Function],
            "title": "Logout!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
